panther_lights/src/animation/animation.py

```diff
@@ -34,11 +36,35 @@ def __init__(self, animation_description: dict, num_led: int, controller_freq: f
                 raise KeyError('Brightness has to match boundaries 0 < brightness <= 1')
             self._brightness = int(round(self._brightness * 100))
 
-    def __call__(self) -> None:
-        '''returns new frame'''
-        raise NotImplementedError
+        # evaluate number of animation frames
+        self._anim_len = int(round(self._duration * controller_freq))
+        self._full_anim_len = self._anim_len * self._loops
+
+    def __call__(self) -> list:
+        if self._current_cycle <= self._loops:
+            frame = self._update_frame()
+            self._anim_iteration += 1
+            self._progress = (
+                self._anim_iteration + self._anim_len * (self._current_cycle - 1)
+            ) / self._full_anim_len
+
+            if self._anim_iteration == self._anim_len:
```

Comment:
```suggestion
            if self._anim_iteration >= self._anim_len:
```
Just for the sake of standardization